meta-resin-common/recipes-bsp/u-boot/patches/resin-specific-env-configuration.patch

```diff
@@ -35,75 +35,80 @@ index 0000000..dfef80c
-+		"echo Scanning MMC devices ${resin_uboot_mmc_devices};" \
-+		"for resin_scan_mmcdev in ${resin_uboot_mmc_devices}; do " \
-+			"if run resin_flasher_detect; then " \
-+				"setenv resin_flasher_mmcdev ${resin_scan_mmcdev}; " \
-+				"exit; " \
-+			"else; " \
-+				"if test -n \"${resin_image_mmcdev}\"; then ;" \
-+					"else if run resin_image_detect; then " \
-+						"setenv resin_image_mmcdev ${resin_scan_mmcdev}; " \
-+					"fi; " \
-+				"fi; " \
-+			"fi; " \
-+		"done;\0" \
-+	\
-+	"resin_set_mmcdev=" \
-+		"run resin_scan_mmcdevs; " \
-+		"if test -n ${resin_flasher_mmcdev}; then " \
-+			"echo Found resin flasher on MMC ${resin_flasher_mmcdev}; "\
-+			"setenv resin_mmcdev ${resin_flasher_mmcdev}; "\
-+		"else; "\
-+			"if test -n \"${resin_image_mmcdev}\"; then " \
-+				"echo Found resin image on MMC ${resin_image_mmcdev}; "\
-+				"setenv resin_mmcdev ${resin_image_mmcdev}; "\
-+			"else; " \
-+				"echo ERROR: Could not find a resin image of any sort.; " \
-+			"fi; " \
-+		"fi;\0" \
-+	\
-+	"resin_inject_env_file=" \
-+		"if run resin_load_env_file; then " \
-+			"run resin_import_env_file;" \
-+		"fi;\0" \
-+	\
-+	"resin_set_kernel_root=" \
-+		"run resin_set_mmcdev;" \
-+		"run resin_inject_env_file;" \
-+		"run resin_find_root_part_uuid;" \
-+		"setenv resin_kernel_root root=PARTUUID=${resin_root_part_uuid}\0"
++       "resin_env_file=" __stringify(RESIN_ENV_FILE) "\0" \
++       "resin_flasher_flag_file=" __stringify(RESIN_FLASHER_FLAG_FILE) "\0" \
++       "resin_image_flag_file=" __stringify(RESIN_IMAGE_FLAG_FILE) "\0" \
++       "resin_uboot_devices=" __stringify(RESIN_UBOOT_DEVICES) "\0" \
++       "resin_boot_part=" __stringify(RESIN_BOOT_PART) "\0" \
++       "resin_root_part=" __stringify(RESIN_DEFAULT_ROOT_PART) "\0" \
++       \
++       "resin_find_root_part_uuid=" \
++               "part uuid ${resin_dev_type} ${resin_dev_index}:${resin_root_part} resin_root_part_uuid\0" \
++       \
++       "resin_load_env_file=" \
++               "echo Loading ${resin_env_file} from ${resin_dev_type} device ${resin_dev_index} partition ${resin_boot_part};" \
++               "fatload ${resin_dev_type} ${resin_dev_index}:${resin_boot_part} ${resin_kernel_load_addr} ${resin_env_file};\0" \
++       \
++       "resin_import_env_file=" \
++               "echo Import ${resin_env_file} in environment;" \
++               "env import -t ${resin_kernel_load_addr} ${filesize}\0" \
++       \
++       "resin_flasher_detect=" \
++               "fatload ${resin_scan_dev_type} ${resin_scan_dev_index}:${resin_boot_part} ${resin_kernel_load_addr} ${resin_flasher_flag_file};\0" \
++       \
++       "resin_image_detect=" \
++               "fatload ${resin_scan_dev_type} ${resin_scan_dev_index}:${resin_boot_part} ${resin_kernel_load_addr} ${resin_image_flag_file};\0" \
++       \
++       "resin_scan_devs=" \
++               "echo Scanning MMC devices ${resin_uboot_devices} and USB devices ${resin_uboot_devices};" \
```

Comment:
I would change this to something like scanning mmc and usb devices ${resin_uboot_devices}